testsuite/tests/publish/hash-internals/test.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+"""
+Test folder hashing internals (order, items, contents)
+
+The following folder structure is being hashed in this test:
+
+./a		# file
+./b		# dir
+./b/x		# file
+./b/y		# dir
+./b/y/p		# file
+./b/y/q		# file
+./b/z		# file
+./á		# utf8-encoded file
+./c		# file
+
+"""
+
+from glob import glob
+
+from drivers.alr import run_alr
+from drivers.asserts import assert_match
+
+import re
+
+# Get the release and check hashing log messages
+p = run_alr('get', '-vv', 'libfoo',
+            complain_on_error=True, quiet=False)
+assert_match(
```

Comment:
It seems a bit dangerous to base the test on debug output.

Reply:
I haven’t looked at the testcase, but generally: IMO it’s better to avoid relying on debug output as much as possible. That being said, when it’s not possible, it’s better to rely on debug output to write a test compared to not writing a test at all. :-) Worst case is that the debug output changes and the testcase needs to be updated, creating maintainance burden. To arbitrate with the importance of the test…